features/gui/admin_panel/lua_console.lua

```diff
@@ -0,0 +1,116 @@
+local Gui = require 'utils.gui'
+local Global = require 'utils.global'
+local AdminPanel = require 'features.gui.admin_panel.core'
+
+local main_button_name = Gui.uid_name()
+local clear_button_name = Gui.uid_name()
+local dry_run_button_name = Gui.uid_name()
+local confirm_button_name = Gui.uid_name()
+
+local this = {
+  last_lua_input = {},
+  last_lua_output = {},
+}
+
+Global.register(this, function(tbl) this = tbl end)
+
+local pages = AdminPanel.get_pages()
+pages[#pages +1] = {
+  type = 'sprite-button',
+  sprite = 'utility/scripting_editor_icon',
+  tooltip = '[font=default-bold]Lua console[/font]',
+  name = main_button_name,
+  auto_toggle = true,
+}
+
+local function draw_gui(player)
+  local canvas = AdminPanel.get_canvas(player)
+  canvas.clear()
```

Comment:
```suggestion
  Gui.clear(canvas)
```
Basically whenever you use `Gui.set_data` you need to use `Gui.destroy` and `Gui.clear` instead of `destory` and `clear` else you get a memory leak.

Reply:
I see there are other places where you have used `.clear()` so will need to change those too.

Reply:
By the way you can check for memory leaks, by seeing what gui data is active for each player using the `/debug` command and then under Gui Data.

Reply:
Whoa 😮 I didnt know we had such an amazing tool! Definitely need to learn more about all our debug options 
![Screenshot from 2024-09-01 10-54-16](https://github.com/user-attachments/assets/d8ced9df-cd7a-4976-b04b-3f1f3af1015b)